src/docker/restorableappengine.cc

```diff
@@ -92,6 +108,32 @@ void RestorableAppEngine::pullAppImages(const boost::filesystem::path& app_compo
   }
 }
 
+void RestorableAppEngine::installApp(const boost::filesystem::path& app_dir, const boost::filesystem::path& dst_dir) {
+  const Manifest manifest{Utils::parseJSONFile(app_dir / Manifest::Filename)};
+  const auto archive_full_path{app_dir / (HashedDigest(manifest.archiveDigest()).hash() + Manifest::ArchiveExt)};
+
+  boost::filesystem::create_directories(dst_dir);
+  auto cmd = boost::str(boost::format("tar --overwrite -xzf %s") % archive_full_path.string());
+  if (0 != boost::process::system(cmd, boost::process::start_dir = dst_dir)) {
+    throw std::runtime_error("failed to install Compose App: " + app_dir.string());
+  }
+}
+
+void RestorableAppEngine::installAppImages(const boost::filesystem::path& app_dir) {
+  const auto compose{ComposeInfo((app_dir / ComposeFile).string())};
+  for (const auto& service : compose.getServices()) {
+    const auto image_uri = compose.getImage(service);
+    LOG_ERROR << image_uri;
+
+    const Uri uri{Uri::parseUri(image_uri)};
+    const std::string tag{uri.registryHostname + '/' + uri.repo + ':' + uri.digest.shortHash()};
```

Comment:
This is just an image tag. It doesn't play any functional role, it could be an empty as it is now or any other random value.
Unfortunately, `skopeo` doesn't allow to specify "destination" by using the full sha256  (@sha256:<>), what actually leads to unnecessary fetch of  a newly exported/loaded image manifest by the docker daemon regardless of the image presence in the docker store. It's not critical since just a manifest is fetched, but I'll try to fix/improve on the next iteration, it will require patching of the skopeo or even dockers source code.

```
var/rootdirs/home/fio# cat /var/lib/docker/image/overlay2/repositories.json | jq
{
  "Repositories": {
    "hub.foundries.io/msul-dev01/app-05": {
      "hub.foundries.io/msul-dev01/app-05:5f80d09": "sha256:36264a214873ebd01966c95be4c53ae879f67361a5ae5c2b89dc14faf3b0b760",
      "hub.foundries.io/msul-dev01/app-05@sha256:5f80d09611ee4f0ba07e7318f9c2970a2d8d0ca84e187e99eae971189a207a71": "sha256:36264a214873ebd01966c95be4c53ae879f67361a5ae5c2b89dc14faf3b0b760"
    }
  }
}

```   
The first record `"hub.foundries.io/msul-dev01/app-05:5f80d09"` is added by aklite/skopeo, the second is added by dockerd. Once we teach skopeo/dockerd to accept the full sha256 hash (not just a tag) then we will eliminate the unnecessary image manifest pull.